routing/router.go

```diff
@@ -170,6 +170,9 @@ type ChannelRouter struct {
 	routeCacheMtx sync.RWMutex
 	routeCache    map[routeTuple][]*Route
 
+	diamCacheMtx  sync.RWMutex
+	diamCache     uint8
```

Comment:
Why is `diamCache` a `uint8`? It's definitely within the realm of possibility (however unlikely) for the network to have a diameter greater than 255. There's not really any loss in efficiency to just make this a `uint32`.